bson-kotlinx/src/main/kotlin/org/bson/codecs/kotlinx/DateTimeSerializers.kt

```diff
@@ -0,0 +1,217 @@
+ * [kotlinx.datetime.Instant.fromEpochMilliseconds] and stored to millisecond accuracy.
+ *
+ * @since 5.2
+ */
+@ExperimentalSerializationApi
+public object InstantAsBsonDateTime : KSerializer<Instant> {
+    override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("InstantAsBsonDateTime", PrimitiveKind.STRING)
+
+    override fun serialize(encoder: Encoder, value: Instant) {
+        when (encoder) {
+            is BsonEncoder -> encoder.encodeBsonValue(BsonDateTime(value.toEpochMilliseconds()))
+            else -> throw SerializationException("Instant is not supported by ${encoder::class}")
+        }
+    }
+
+    override fun deserialize(decoder: Decoder): Instant {
+        return when (decoder) {
+            is BsonDecoder -> Instant.fromEpochMilliseconds(decoder.decodeBsonValue().asDateTime().value)
+            else -> throw SerializationException("Instant is not supported by ${decoder::class}")
+        }
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    public val serializersModule: SerializersModule = SerializersModule {
+        contextual(Instant::class, InstantAsBsonDateTime as KSerializer<Instant>)
+    }
+}
+
+/**
+ * LocalDate KSerializer.
+ *
+ * <p>Encodes and decodes {@code LocalDate} objects to and from {@code BsonDateTime}.</p>
+ *
+ * <p>Converts the {@code LocalDate} values to and from {@code UTC}.</p>
+ *
+ * @since 5.2
+ */
+@ExperimentalSerializationApi
+public object LocalDateAsBsonDateTime : KSerializer<LocalDate> {
+    override val descriptor: SerialDescriptor =
+        PrimitiveSerialDescriptor("LocalDateAsBsonDateTime", PrimitiveKind.STRING)
+
+    override fun serialize(encoder: Encoder, value: LocalDate) {
+        when (encoder) {
+            is BsonEncoder -> {
+                val epochMillis = value.atStartOfDayIn(TimeZone.UTC).toEpochMilliseconds()
+                encoder.encodeBsonValue(BsonDateTime(epochMillis))
+            }
+            else -> throw SerializationException("LocalDate is not supported by ${encoder::class}")
+        }
+    }
+
+    override fun deserialize(decoder: Decoder): LocalDate {
+        return when (decoder) {
+            is BsonDecoder ->
+                Instant.fromEpochMilliseconds(decoder.decodeBsonValue().asDateTime().value)
+                    .toLocalDateTime(TimeZone.UTC)
+                    .date
+            else -> throw SerializationException("LocalDate is not supported by ${decoder::class}")
+        }
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    public val serializersModule: SerializersModule = SerializersModule {
+        contextual(LocalDate::class, LocalDateAsBsonDateTime as KSerializer<LocalDate>)
+    }
+}
+
+/**
+ * LocalDateTime KSerializer.
+ *
+ * <p>Encodes and decodes {@code LocalDateTime} objects to and from {@code BsonDateTime}. Data is stored to millisecond
+ * accuracy.</p>
+ *
+ * <p>Converts the {@code LocalDateTime} values to and from {@code UTC}.</p>
+ *
+ * @since 5.2
```

Comment:
```suggestion
 * Encodes and decodes `LocalDateTime` objects to and from `BsonDateTime`. Data is stored to millisecond
 * accuracy.
 *
 * Converts the `LocalDateTime` values to and from `UTC`.
 *
 * @since 5.2
```